### ISSUE TITLE
Moodle STACK connect Question Text and Question Variables (Maxima)

### ISSUE THREAD

**Issue by BlackBlueGreens** [IDIAMath/idiamath.github.io#26]

**Comment by BlackBlueGreens** [comment]
The problem could very well be for example when you perform diff on funcExpr2, say your input is x*y and so funcExpr2 is x*y, but when you perform diff it performs it on quit literally funcExpr2 instead of x*y which is what it is equal to.

**Comment by jonasjul** [comment]
It doesn't look like the value of funcExpr2 is sent to maxima - it simply lives as a javascript variable. You shouldn't need to update it anyway: The function the student gives is stored in "ans1", and is available to maxima when the answer is graded.

**Comment by BlackBlueGreens** [comment]
Correct. It doesn't seem like its possible to achieve what I am trying to achieve, since i want maxima to run before the answer is submitted. The idea was that when the student clicks **'Draw'**, JS takes whatever function from the input field and draw it, along side some of its partial derivatives. We can only find its partial derivative's via operations in maxima.

**Comment by jonasjul** [comment]
We should have some means for the students to investigate the partial derivatives of the function they are tasked to find. Automatically computing the partial derivatives in maxima and plotting them when the student supplies a function would have been very neat - But I don't see a way to call maxima WHILE the student is working on the question. After submission we can work on the students function with maxima in the "feedback variables" field; Which means perhaps can show an interactive plot from the feedback field in the response tree (I have not checked if jsxgraph-blocks work here).  If we change the question behaviour to "interactive with multiple tries" this could perhaps accomplish what we want.
 
Alternatively, it could perhaps be enough to have jsxgraph also show the tangent plane at a dragable point in the domain - This also eludes to /shows the sign of the partial derivatives. Check out the "3d function graph" example found here: https://jsxgraph.uni-bayreuth.de/~alfred/jsxdev/3D/demo3d.html
To accomplish this, we would probably need to use finite differences to approximate the partial derivatives:
https://folk.ntnu.no/leifh/teaching/tkt4140/._main007.html
